File: packages/sdk/src/models/PrimerCheckoutAdditionalInfo.ts
export type PrimerCheckoutAdditionalInfo = IPrimerCheckoutAdditionalInfo;

interface IPrimerCheckoutAdditionalInfo {}

export type MultibancoCheckoutAdditionalInfo = IMultibancoCheckoutAdditionalInfo;

interface IMultibancoCheckoutAdditionalInfo extends IPrimerCheckoutAdditionalInfo {
    expiresAt?: string;
    entity?: string;
    reference?: string;
}

export type PrimerCheckoutQRCodeInfo = IPrimerCheckoutQRCodeInfo;

interface IPrimerCheckoutQRCodeInfo extends IPrimerCheckoutAdditionalInfo {}

export type PromptPayCheckoutAdditionalInfo = IPromptPayCheckoutAdditionalInfo;

interface IPromptPayCheckoutAdditionalInfo extends IPrimerCheckoutQRCodeInfo {
    expiresAt: string;
    qrCodeUrl?: string;
    qrCodeBase64?: string;
}

export type PrimerCheckoutVoucherAdditionalInfo = IPrimerCheckoutVoucherAdditionalInfo;

interface IPrimerCheckoutVoucherAdditionalInfo extends IPrimerCheckoutAdditionalInfo {}

export type XenditCheckoutVoucherAdditionalInfo = IXenditCheckoutVoucherAdditionalInfo;

interface IXenditCheckoutVoucherAdditionalInfo extends IPrimerCheckoutVoucherAdditionalInfo {
    expiresAt: string;
    couponCode: string;
    retailerName: string;
}

File: packages/sdk/android/src/main/java/com/primerioreactnative/extensions/PrimerCheckoutAdditionalInfo.kt
package com.primerioreactnative.extensions

import com.primerioreactnative.datamodels.AchAdditionalInfoDisplayMandateRN
import com.primerioreactnative.datamodels.MultibancoCheckoutAdditionalInfoRN
import com.primerioreactnative.datamodels.PromptPayCheckoutAdditionalInfoRN
import com.primerioreactnative.datamodels.XenditCheckoutVoucherAdditionalInfoRN
import io.primer.android.payments.core.additionalInfo.PrimerCheckoutAdditionalInfo
import io.primer.android.qrcode.QrCodeCheckoutAdditionalInfo
import io.primer.android.stripe.ach.api.additionalInfo.AchAdditionalInfo
import io.primer.android.vouchers.multibanco.MultibancoCheckoutAdditionalInfo
import io.primer.android.vouchers.retailOutlets.XenditCheckoutVoucherAdditionalInfo

internal fun PrimerCheckoutAdditionalInfo.toCheckoutAdditionalInfoRN() =
  when (this) {
    is MultibancoCheckoutAdditionalInfo ->
      MultibancoCheckoutAdditionalInfoRN(expiresAt, reference, entity)
    is QrCodeCheckoutAdditionalInfo ->
      PromptPayCheckoutAdditionalInfoRN(expiresAt.orEmpty(), qrCodeUrl, qrCodeBase64)
    is XenditCheckoutVoucherAdditionalInfo ->
      XenditCheckoutVoucherAdditionalInfoRN(expiresAt, couponCode, retailerName)
    is AchAdditionalInfo.DisplayMandate ->
      AchAdditionalInfoDisplayMandateRN()
    else -> null
  }
